Sources/Attribution/AttributionFetcher.swift

```diff
@@ -67,35 +73,22 @@ class AttributionFetcher {
         return nil
     }
 
-    func afficheClientAttributionDetails(completion: @escaping ([String: NSObject]?, Error?) -> Void) {
-        // Should match available platforms in
-        // https://developer.apple.com/documentation/iad/adclient?language=swift
-#if os(iOS)
-        guard let afficheClientProxy = attributionFactory.afficheClientProxy() else {
-            Logger.warn(Strings.attribution.search_ads_attribution_cancelled_missing_ad_framework)
-            completion(nil, AttributionFetcherError.identifierForAdvertiserFrameworksUnavailable)
-            return
+    // should match OS availability in https://developer.apple.com/documentation/ad_services
+    @available(iOS 14.3, macOS 11.1, macCatalyst 14.3, *)
+    func adServicesToken() -> String? {
+#if canImport(AdServices)
+        if let attributionToken = try? AAAttribution.attributionToken() {
+            return attributionToken
+        } else {
+            Logger.warn(Strings.attribution.adservices_token_fetch_failed)
```

Comment:
Maybe do a do/catch instead of `try?` to add the underlying error to this?
Also I wonder if it should be `Logger.appleError`